src/Microsoft.AspNetCore.Razor.Language/Syntax/Syntax.xml

```diff
@@ -0,0 +1,102 @@
+    </Field>
+    <Field Name="EndCommentStar" Type="SyntaxToken">
+      <Kind Name="RazorCommentStar" />
+    </Field>
+    <Field Name="EndCommentTransition" Type="SyntaxToken">
+      <Kind Name="RazorCommentTransition" />
+    </Field>
+  </Node>
+
+  <!-- HTML -->
+  <AbstractNode Name="HtmlNodeSyntax" Base="RazorNodeSyntax" />
+  <Node Name="HtmlTextSyntax" Base="HtmlNodeSyntax">
+    <Kind Name="HtmlText" />
+    <Field Name="TextTokens" Type="SyntaxList&lt;SyntaxToken&gt;" />
+  </Node>
+  
+  <!-- CSharp -->
+  <AbstractNode Name="CSharpNodeSyntax" Base="RazorNodeSyntax" />
+  <Node Name="CSharpTransitionSyntax" Base="CSharpNodeSyntax">
+    <Kind Name="CSharpTransition" />
+    <Field Name="Transition" Type="SyntaxToken">
+      <Kind Name="Transition" />
+    </Field>
+  </Node>
+  <Node Name="CSharpMetaCodeSyntax" Base="CSharpNodeSyntax">
+    <Kind Name="CSharpMetaCode" />
+    <Field Name="MetaCode" Type="SyntaxList&lt;SyntaxToken&gt;" />
+  </Node>
+  <Node Name="CSharpCodeLiteralSyntax" Base="CSharpNodeSyntax">
+    <Kind Name="CSharpCodeLiteral" />
+    <Field Name="CSharpTokens" Type="SyntaxList&lt;SyntaxToken&gt;" />
+  </Node>
+  <Node Name="CSharpCodeBlockSyntax" Base="CSharpNodeSyntax">
+    <Kind Name="CSharpCodeBlock" />
+    <Field Name="Children" Type="SyntaxList&lt;RazorNodeSyntax&gt;" />
+  </Node>
+  <AbstractNode Name="CSharpBlockSyntax" Base="CSharpNodeSyntax">
+    <Field Name="Transition" Type="CSharpTransitionSyntax" />
+    <Field Name="Body" Type="CSharpNodeSyntax" />
+  </AbstractNode>
+  <Node Name="CSharpStatement" Base="CSharpBlockSyntax">
+    <Kind Name="CSharpStatement" />
+    <Field Name="Transition" Type="CSharpTransitionSyntax" Override="true">
+      <Kind Name="CSharpTransition" />
+    </Field>
+    <Field Name="Body" Type="CSharpNodeSyntax" Override="true">
+      <Kind Name="CSharpStatementBody" />
+    </Field>
+  </Node>
+  <Node Name="CSharpStatementBodySyntax" Base="CSharpNodeSyntax">
+    <Kind Name="CSharpStatementBody" />
+    <Field Name="OpenBrace" Type="CSharpMetaCodeSyntax" />
+    <Field Name="CSharpCode" Type="CSharpCodeBlockSyntax" />
+    <Field Name="CloseBrace" Type="CSharpMetaCodeSyntax" />
+  </Node>
+  <Node Name="CSharpExpression" Base="CSharpBlockSyntax">
+    <Kind Name="CSharpExpression" />
+    <Field Name="Transition" Type="CSharpTransitionSyntax" Override="true">
+      <Kind Name="CSharpTransition" />
+    </Field>
+    <Field Name="Body" Type="CSharpNodeSyntax" Override="true">
+      <Kind Name="CSharpExpressionBody" />
+    </Field>
+  </Node>
+  <Node Name="CSharpExpressionBodySyntax" Base="CSharpNodeSyntax">
+    <Kind Name="CSharpExpressionBody" />
+    <Field Name="OpenParen" Type="CSharpMetaCodeSyntax" Optional="true" />
+    <Field Name="CSharpCode" Type="CSharpCodeBlockSyntax" />
+    <Field Name="CloseParen" Type="CSharpMetaCodeSyntax" Optional="true" />
+  </Node>
+  <Node Name="CSharpDirectiveSyntax" Base="CSharpBlockSyntax">
```

Comment:
Lets not change anything now but just re-iterating cases such as `@page` are 0% C# and would be nice to eventually parse them outside of C#.

For now, what do you think about changing all of the `Razor` syntax concepts to not include C# even though they're parsed in the C# parser?

Reply:
> For now, what do you think about changing all of the Razor syntax concepts to not include C# even though they're parsed in the C# parser?

I prefer doing that later as well as this might make it confusing when trying to convert the current parser. It should be easy to do once we have the parser actually working with the new nodes.